doc/parallel.md

```diff
@@ -0,0 +1,67 @@
+
+```c++
+  std::function<void(const DispenserModel &)> run(Dispenser &sut) const override {
+    auto ticket = sut.takeTicket();
+
+    return [ticket](const DispenserModel &model) {
+      RC_ASSERT(ticket == (model.ticket + 1));
+    };
+  }
+```
+
+Finally, use the checkParallel function to execute the test in parallel mode.
+
+```c++
+  rc::check([] {
+    DispenserModel initialState;
+    Dispenser sut;
+    rc::state::checkParallel(initialState, sut,
+                             rc::state::gen::execOneOf<Take, Reset>);
+  });
+```
+
+## Output ##
+The output from a parallel test looks a bit different from a sequntial test.
+
+```
+Falsifiable after 97 tests and 3 shrinks
+
+Parallel command sequence of Command<DispenserModel, Dispenser>:
+Sequential prefix:
+Reset
+Left branch:
+Take
+Right branch:
+Reset
+Take
+```
+
+In the example above we can see the base type of the commands in the test```Parallel command sequence of Command<DispenserModel, Dispenser>:```. This is followed by the three sections ```Sequential prefix```, ```Left branch``` and ```Right branch``` which contains the commands that were executed in order to reproduce the bug. The prefix is a sequence of commands that were executed sequentially before any of the commands in the branches were executed. The left and right branches contains commands that were executed in parallel.
+
+## Tips ##
+### Shrinking parallel tests ###
+Race conditions are often not deterministically reproducible. If this is the case you can easily end up with a failing test that RapidCheck is not able to shrink; the probability of the race condition being triggered when testing each shrink is simply too low. By using the configuration parameter ```shrink_tries``` you can increase the probability of the race condtion being triggered by increasing the number of times each shrink is tested.
+
+### Be careful when using the command constructor that takes the model as an argument ###
+For a sequential test the order in which commands are generated is the same as the order in which the commands will be executed. This is not the case in the parallel case however. The commands in the parallel branches are provided a model that is created by invoking the apply function for each command in the prefix, followed by each preceding command in the branch. This is generally not the same order that the commands will be executed in. In some cases this makes creating commands for parallel tests more challanging, since you cannot assume that the model that is provided to your command constructor reflects the state of the model at the time of the execution of the command.
+
+Note however that this restriction does not apply to the ```apply``` function. Wheras each command is only created once, the apply function is executed once for every possible interleaving (at least conceptually).
```

Comment:
Fixed in 114ae6fdd719d2acfe3769cb313661095ba330c8. Thanks for the backtick tip...